File: phonegraph-llm-orchestrator/src/main/java/com/phonegraph/orchestrator/model/RecommendationRequest.java
package com.phonegraph.orchestrator.model;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

/**
 * RecommendationRequest — what the Android app (or a test client) sends
 * when asking PhoneGraph for a phone recommendation.
 */
@Data
public class RecommendationRequest {

    @NotBlank
    private String query;          // e.g. "best camera phone under £400"

    private String brand;          // optional constraint
    private Integer maxPrice;      // optional constraint
    private int limit = 5;         // how many candidate phones to consider
}

File: phonegraph-android/app/src/main/java/com/phonegraph/android/data/RecommendationRequest.kt
package com.phonegraph.android.data

/**
 * Mirrors RecommendationRequest.java on the backend exactly (field names
 * and defaults), so Gson serialises this into a body the orchestrator
 * already understands — no backend changes needed for the Android client.
 */
data class RecommendationRequest(
    val query: String,
    val brand: String? = null,
    val maxPrice: Int? = null,
    val limit: Int = 5
)
